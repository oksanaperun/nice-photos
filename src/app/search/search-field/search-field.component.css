.mat-input-element,
.mat-form-field-placeholder-wrapper {
  font-family: 'Nunito', serif;
}

.mat-input-container {
  margin-left: 5px;
  width: 300px;
}

.mat-input-element {
  font-size: x-large;
  color: rgba(0,0,0,.87);;
}
