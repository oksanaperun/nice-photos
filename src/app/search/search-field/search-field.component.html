<mat-form-field>
  <input matInput placeholder="Search nice photos" [formControl]="searchInput"
         [errorStateMatcher]="matcher">
  <mat-error *ngIf="searchInput.hasError('required')">
    Search text is <strong>required</strong>
  </mat-error>
  <mat-error *ngIf="searchInput.hasError('pattern')">
    Special characters or just spaces are <strong>not allowed</strong>
  </mat-error>
</mat-form-field>
