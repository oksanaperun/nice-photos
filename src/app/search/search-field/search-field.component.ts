import { Component, ChangeDetectionStrategy } from '@angular/core';
import { FormControl, Validators } from '@angular/forms';
import { SearchTextErrorStateMatcher } from './searchTextErrorStateMatcher';

@Component({
  selector: 'app-search-field',
  templateUrl: './search-field.component.html',
  styleUrls: ['./search-field.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})

export class SearchFieldComponent {
  searchInput = new FormControl('', [
    Validators.required,
    Validators.pattern('((?=.*[a-zA-Z])|(?=.*[0-9]))[a-zA-Z0-9 ]+')
  ]);
  matcher = new SearchTextErrorStateMatcher();
}
