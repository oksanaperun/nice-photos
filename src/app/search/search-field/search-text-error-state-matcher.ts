import { ErrorStateMatcher } from '@angular/material/core';
import { FormControl, NgForm } from '@angular/forms';

export class SearchTextErrorStateMatcher implements ErrorStateMatcher {
  isErrorState(control: FormControl | null, form: NgForm | null): boolean {
    const isSubmitted = form && form.submitted;

    return !!(control && control.invalid && (control.dirty || control.touched || isSubmitted));
  }
}
