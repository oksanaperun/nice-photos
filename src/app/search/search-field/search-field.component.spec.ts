import { TestBed, ComponentFixture } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { SearchFieldComponent } from './search-field.component';

describe('SearchFieldComponent', () => {
  let component: SearchFieldComponent;
  let fixture: ComponentFixture<SearchFieldComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [SearchFieldComponent],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();

    fixture = TestBed.createComponent(SearchFieldComponent);
    component = fixture.componentInstance;
  });

  it('should NOT render error when input is valid', () => {
    component.searchInput.setValue('abc');

    expect(fixture).toMatchSnapshot();
  });

  it('should render error when input contains special character', () => {
    component.searchInput.setValue('ab,c');
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should render error when input contains only spaces', () => {
    component.searchInput.setValue('  ');
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should render error when input is empty', () => {
    component.searchInput.setValue('');
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
