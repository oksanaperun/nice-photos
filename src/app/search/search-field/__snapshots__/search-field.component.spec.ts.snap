// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFieldComponent should NOT render error when input is valid 1`] = `
<app-search-field
  matcher={[Function SearchTextErrorStateMatcher]}
  searchInput={[Function FormControl]}
>
  <mat-form-field>
    
    
    <input
      matinput=""
      placeholder="Search nice photos"
    />
    
    
    
    
    
    
    
  
  </mat-form-field>
  
</app-search-field>
`;

exports[`SearchFieldComponent should render error when input contains only spaces 1`] = `
<app-search-field
  matcher={[Function SearchTextErrorStateMatcher]}
  searchInput={[Function FormControl]}
>
  <mat-form-field>
    
    
    <input
      matinput=""
      placeholder="Search nice photos"
    />
    
    
    
    
    
    
    <mat-error>
      
      Special characters or just spaces are 
      <strong>
        not allowed
      </strong>
      
    
    </mat-error>
    
  
  </mat-form-field>
  
</app-search-field>
`;

exports[`SearchFieldComponent should render error when input contains special character 1`] = `
<app-search-field
  matcher={[Function SearchTextErrorStateMatcher]}
  searchInput={[Function FormControl]}
>
  <mat-form-field>
    
    
    <input
      matinput=""
      placeholder="Search nice photos"
    />
    
    
    
    
    
    
    <mat-error>
      
      Special characters or just spaces are 
      <strong>
        not allowed
      </strong>
      
    
    </mat-error>
    
  
  </mat-form-field>
  
</app-search-field>
`;

exports[`SearchFieldComponent should render error when input is empty 1`] = `
<app-search-field
  matcher={[Function SearchTextErrorStateMatcher]}
  searchInput={[Function FormControl]}
>
  <mat-form-field>
    
    
    <input
      matinput=""
      placeholder="Search nice photos"
    />
    
    
    
    <mat-error>
      
      Search text is 
      <strong>
        required
      </strong>
      
    
    </mat-error>
    
    
    
    
  
  </mat-form-field>
  
</app-search-field>
`;
